### PR TITLE
Add lazy loading to skimage.restoration

### PR DESCRIPTION
**Pull request by ana42742**
## Description
Resolves a task in #6964

## Checklist

- A descriptive but concise pull request title
- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [Unit tests](https://scikit-image.org/docs/dev/development/contribute.html#testing)
- A gallery example in `./doc/examples` for new features
- [Contribution guide](https://scikit-image.org/docs/dev/development/contribute.html) is followed

### CODE REVIEW

File: skimage/restoration/__init__.pyi [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+           'denoise_tv_chambolle',
+           'denoise_bilateral',
+           'denoise_wavelet',
+           'denoise_nl_means',
+           'denoise_invariant',
+           'estimate_sigma',
+           'inpaint_biharmonic',
+           'cycle_spin',
+           'calibrate_denoiser',
+           'rolling_ball',
```
Comment (lagru) [inline skimage/restoration/__init__.pyi:19]: Looks good,thanks. I think [on MacOS this function is again confused with the submodule of the same name](https://github.com/scikit-image/scikit-image/actions/runs/5253063227/jobs/9489909042#step:6:536). :roll_eyes: Could you prefix `skimage/restoration/rolling_ball.py` accordingly with a `_` like in previous PRs?

Comment (lagru) [reply skimage/restoration/__init__.pyi:19]: And feel welcome to do the same for the other "public" submodules in `skimage/restoration`. They should be private.

Comment (ana42742) [reply skimage/restoration/__init__.pyi:19]: Sure, I will update and refactor changes.

**Review by lagru — APPROVED** [review]
Looks good, the test failure is unrelated (see https://github.com/scikit-image/scikit-image/issues/7027). Thanks @ana42742!

**Review by jarrodmillman — APPROVED** [review]

**Pull request merged**